### DISCUSSION TITLE
Need help to understand the Thompson Sampling (TS) implementation in the TuRBO tutorial

### DISCUSSION THREAD

**Discussion by Ted12345678** [meta-pytorch/botorch#1786] [Q&A]
I am looking into the TuRBO tutorial https://botorch.org/tutorials/turbo_1, I am having a hard time to understand the perturbation mask. For example, how is 20.0 is determined in the following code?

        # Create a perturbation mask
        prob_perturb = min(20.0 / dim, 1.0)

Any theory or tutorial about the perturbation mask and its usage in TS implementation?

Thank you very much in advance!

**Comment by Balandat** [comment]
@dme65, @sdaulton can you help with this?
